### ISSUE TITLE
[Feature request] Control optionality via separate rule

### ISSUE THREAD

**Issue by gryznar** [astral-sh/ty#3843]
## Intro

`ty` currently treats completely invalid operations on a type with the same severity as operations that are merely missing a `None` check. This conflates critical bugs (typos/type mismatches) with null-safety violations.

```python
# Case 1: Invalid Operation (always fails)
def foo(bar: int):
    bar.to_bytess()  # Typo: 'int' has no such method

# Case 2: Nullability Violation (potential error)
def foo(bar: int | None):
    bar.to_bytes()   # Valid method for 'int', but 'bar' could be None
```

## Proposed Feature

Separate rule to control optionality checks.

## Benefit

This will allow to set nullability check as "warn", but keep rest of rules as "error".

### ​Immediate Bug Detection

Catches typos and structural errors instantly without them being obscured by strict null-safety checks.

### ​Incremental Adoption

Allows treating missing `None` guards as non-blocking warnings while keeping strict type enforcement for typos.

**Comment by AlexWaygood** [comment]
Thanks for the feature request. This is https://github.com/astral-sh/ty/issues/2839
